.yamato/scripts/update_from_upstream.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+# Prequisites:
+# This script assumes git and github CLR (gh) are installed.
+
+# Cause the script to fail if any individual command fails
+set -e
+
+# Fetch from upstream and rebase our local main on it
+git remote add upstream https://github.com/dotnet/runtime
+git fetch upstream
+git checkout --track origin/main
+git merge --ff-only upstream/main
+
+# Push our merged local main to origin
+git push
+
+# Create a new branch for a pull request to merge in main
+# using the current date in the branch name.
+git checkout unity-main
+branch_name=bot-upstream-main-merge-$(date '+%Y-%m-%d')
+git checkout -b $branch_name
+git merge main -m "Merge with main from upstream"
+git push --set-upstream origin $branch_name
+
+# Login to Github
+set -u
```

Comment:
Do we need to create a PR here?  We'd already fail at https://github.com/Unity-Technologies/runtime/pull/8/files#diff-ef7c119a3b7f6122d886f3c65142eafe9dd9fd0edaadbfeeb675490477aab298R23 if the merge fails.  Could we run the tests as part of the yamato job and merge the results automatically if it passes?  Or do we want the manual review of the changes?

Reply:
> Could we run the tests as part of the yamato job and merge the results automatically if it passes?

This might be possible. I could see:

1. Yamato job to do the fetch from upstream
2. Yamato job to run the tests
3. Yamato job to merge the branch into unity-main

But this seemed like a simpler workflow.

> Or do we want the manual review of the changes?

I kind of did want a manual review, but looking at the test pull requests, they are probably too big to review.

My preference is to do this simpler set up now. If the pull request review/merge becomes rote, then we can change the process have the machines do it for us later.